### ISSUE TITLE
orca is sloes down on large files

### ISSUE THREAD

**Issue by alex19EP** [microsoft/vscode#140224]
Issue Type: <b>Bug</b>

on large files eg: 2000+ lines orca will become slow on 200+ line
orca will become fast on 501+ line

CC: @isidorn @joanmarie

VS Code version: Code - Insiders 1.64.0-insider (9afbea7caab9dd06592ddd98aa4161b43b126d3f, 2022-01-06T05:15:46.948Z)
OS version: Linux x64 5.15.13-zen1-1-zen
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 x 4200)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 1, 1|
|Memory (System)|15.49GB (0.16GB free)|
|Process Argv|--crash-reporter-id 49edb027-d44c-4fb4-befd-53de5d4dd885|
|Screen Reader|yes|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (64)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|2.1.0
meson|asa|1.3.0
github-markdown-preview|bie|0.1.0
markdown-checkbox|bie|0.3.2
markdown-emoji|bie|0.2.1
markdown-preview-github-styles|bie|0.2.0
markdown-yaml-preamble|bie|0.1.0
better-toml|bun|0.3.2
vscode-astyle|chi|0.9.0
systemd-unit-file|coo|1.0.6
vscode-github-actions|csc|0.21.3
vscode-markdownlint|Dav|0.45.0
githistory|don|0.6.19
xml|Dot|2.5.1
gitlens|eam|11.7.0
EditorConfig|Edi|0.16.4
git-project-manager|fel|1.8.2
remotehub|Git|0.20.0
remotehub-insiders|Git|2021.12.160108
vscode-pull-request-github|Git|0.35.2022010609
gitlab-workflow|Git|3.40.1
markdown-extended|jeb|1.1.1
better-cpp-syntax|jef|1.15.10
restructuredtext|lex|167.0.0
kconfig|luv|1.0.3
rust-analyzer|mat|0.2.867
vscode-github-actions|me-|3.0.1
language-gettext|mro|0.2.0
vscode-docker|ms-|1.18.0
vscode-language-pack-ru|MS-|1.64.2
python|ms-|2022.0.1612153341-dev
vscode-pylance|ms-|2022.1.0
jupyter|ms-|2022.1.1001614873
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.4
remote-containers|ms-|0.210.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
remote-wsl|ms-|0.63.13
cmake-tools|ms-|1.9.2
cpptools|ms-|1.8.0-insiders2
cpptools-extension-pack|ms-|1.1.0
vsliveshare|ms-|1.0.5201
vsliveshare-audio|ms-|0.1.91
vsliveshare-pack|ms-|0.4.0
vsonline|ms-|1.0.3076
awk-vscode|pet|0.1.1
trusty-rusty-snippets|pol|0.0.5
java|red|1.2.0
vscode-commons|red|0.0.6
vscode-yaml|red|1.2.2
platform-settings|run|1.0.0
rewrap|stk|1.15.4
code-spell-checker|str|2.0.14
code-spell-checker-russian|str|2.0.2
even-better-toml|tam|0.14.2
cmake|twx|0.0.17
vscode-lldb|vad|1.6.10
vscodeintellicode|Vis|1.2.15
vscode-java-debug|vsc|0.37.0
vscode-java-dependency|vsc|0.18.9
vscode-java-pack|vsc|0.20.0
vscode-java-test|vsc|0.33.1
vscode-maven|vsc|0.34.2

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
vscop804cf:30404767
vscop341:30404997
vsclayoutctrt:30405800
py55gd98:30408621
azactmsalcf:30409288

```

</details>

**Comment by isidorn** [comment]
@alex19EP thanks for filling this issue. Does changing the setting 
`"editor.accessibilityPageSize"` to a smaller number help here?
Can you share the file for which Orca becomes slow so we try to investigate more?

Thanks!

**Comment by alex19EP** [comment]
> @alex19EP thanks for filling this issue. Does changing the setting `"editor.accessibilityPageSize"` to a smaller number help here?

yes.

> Can you share the file for which Orca becomes slow so we try to investigate more?
> 
> Thanks!

sure
[file.txt](https://github.com/microsoft/vscode/files/7829522/file.txt)

**Comment by isidorn** [comment]
My guess is that this is caused by Chrome bug which makes the `textArea` with a lot of content slow.
Unfortunately there is nothing we can do here, however this is fixed in the next Chrome version, which we will get with the Electron update. We plan to start using the new Electron version in the next couple of months and then this should be fixed.

fyi @deepak1556

**Comment by isidorn** [comment]
This should be fixed now in Insiders since we are using Electron 16.
Closing this optimistically and we can reopen if we decide to go back to Electron 14 for this milestone.

@alex19EP let us know once you try it out.

**Comment by alex19EP** [comment]
still can reproduce this on 1.65.0-insider bb221a61d29deabd99ee9431736d04f2175cb596

**Comment by isidorn** [comment]
Then let's reopen.
@alex19EP would you be so kind to try to reproduce with latest Chrome and https://vscode.dev/github/microsoft/vscode 

If that reproduces, can you also try to reproduce just with a simple text area https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

**Comment by alex19EP** [comment]
> would you be so kind to try to reproduce with latest Chrome and https://vscode.dev/github/microsoft/vscode

of course.
tried in chromium 98.0.4758.102
in microsoft/vscode/src/main.js:499 lag is same as in latest code insiders with electron13

> If that reproduces, can you also try to reproduce just with a simple text area https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

yep I can also reproduce slowdown with this textaria too.

**Comment by isidorn** [comment]
@alex19EP ok, so it seems that the issue is `textarea` + Chrome. Can you file a new issue on the Chromium side https://bugs.chromium.org/p/chromium/issues/list - just mention `textarea` in repro steps, no need for VS Code - so they can repro it easily.
And if you can: share the link to the issue here. Then I can follow up with Google people. Thanks a lot!

**Comment by isidorn** [comment]
@alex19EP thanks for doing that. I see they are asking some questions on the issue. I will try to answer but you should also chime in.

**Comment by isidorn** [comment]
I believe this should work much better with the latest vscode version. Thus I will go ahead and close this issue.
If you still encounter problems do let us know and we can reopen

fyi @meganrogge
